### DISCUSSION TITLE
Accessing private nexus repo

### DISCUSSION THREAD

**Discussion by omega** [JetBrains/Qodana#66] [Q&A] [answered]
Hey,

I am trying to set up qodana with the docker-image, but I get an error when it tries to access our private nexus repositories, and I'm unsure where or how I grant it access to those. Locally I use `.m2/settings.xml` , but can I inject that somehow in the docker-run?

this is the error (host and path of nexus server altered here)

```
java.lang.IllegalStateException: Maven project Fiken Asgeir Server has import problems:[STRUCTURE:authentication failed for https://nexus/content/groups/public/parent/topi-parent/30/topi-parent-30.pom, status: 401 Unauthorized:/data/project/pom.xml]
```

I trid adding `-v/home/<user>/.m2:/.m2` to the docker command line, as a hail marry, but no dice

EDIT: found https://youtrack.jetbrains.com/issue/QD-792, so trying that now

**Comment by tiulpin** [comment, marked as answer]
Hey! Try mapping to `/root/.m2` as `/root` is the default $HOME directory
```
docker run ... -v /home/<user>/.m2:/root/.m2 ... jetbrains/qodana-jvm
``` 
like for Gradle https://www.jetbrains.com/help/qodana/qodana-jvm-docker-techs.html#Provide+different+Gradle+settings

**Reply by omega** [reply]
yeah, found the issue I linked right after posting, so that worked just now, thank for the quick answer as well :)

**Reply by SabareeshGC** [reply]
How can we do this on azure pipeline as it is part of the extension.

**Comment by SabareeshGC** [comment]
How can we do this on azure pipeline as it is part of the extension.

**Comment by jvanzyl** [comment]
Is there a way to configure this using the GitHub action?

**Reply by jcputney** [reply]
The way I ended up getting it to work was with the `whelk-io/maven-settings-xml-action` action to generate my `settings.xml` file in my project root, and then using `bootstrap: if [ -e settings.xml ]; then cp settings.xml ~/.m2/settings.xml; fi` in my `qodana.yml`. The `if` statement is just so it doesn't wipe out my local settings.xml file if I run Qodana locally.
